docs/content/guides/developer/advanced/move-2024-migration.mdx

```diff
@@ -0,0 +1,300 @@
+    let mut a = 0;
+    let S { mut x } = 5;
+    a = 1;
+    x = 10;
+    y = 5;
+    x + y
+}
+```
+
+The compiler now produces an error if you attempt to reassign or borrow a variable mutably without this explicit declaration.
+
+### Removing friends and `public(friend)`
+
+Friends and the `public(friend)` visibilities were introduced early in Move's development, predating even the package system. As indicated in the [public(package)](#public-package) section, `public(package)` deprecates `public(friend)` in Move 2024.
+
+The following declaration now produces an error:
+
+```move
+module pkg::m {
+    friend pkg::a;
+    public(friend) fun f() { ... }
+}
+
+module pkg::a {
+    fun calls_f() { ... pkg::m::f() ... }
+}
+```
+
+Instead, if you want your function to be visible only in the package, write:
+
+```move
+module pkg::m {
+    public(package) fun f() { ... }
+}
+
+module pkg::a {
+    // this now works directly
+    fun calls_f() { ... pkg::m::f() ... }
+}
+```
+
+### New keywords
+
+Looking toward the future, Move 2024 Beta adds the following keywords to the language: `enum`, `for`, `match`, `mut`, and `type`. The compiler, unfortunately, now produces parsing errors when it finds these in other positions. This is a necessary change as the language matures. If you perform automatic migration, the migration tool renames these as `enum` and so on, rewriting the code to use these escaped forms.
+
+### Revised paths and namespaces
+
+Move 2024 revises how paths and namespaces work compared to legacy Move, toward easing `enum` aliasing in the future. Consider the following snippet from a test annotation in the `sui_system` library:
+
+```move
+use sui_system::sui_system;
+...
+#[expected_failure(abort_code = sui_system::validator_set::EInvalidCap)]
+```
+
+Legacy Move would always treat a three-part name as an address(`sui_system`), module(`validator_set`), and module member (`EInvalidCap`). Move 2024 respects scope for `use`, so `sui_system` in the attribute resolves to the module, producing a name resolution error overall.
+
+To avoid cases where this is the intended behavior, Move 2024 introduces a prefix operation for global qualification. To use, you can rewrite this annotation as:
+
+```move
+use sui_system::sui_system;
+...
+#[expected_failure(abort_code = ::sui_system::validator_set::EInvalidCap)]
+                             // ^ note `::` here
+```
+
+The migration script attempts to remediate naming errors using global qualification when possible.
+
+### Follow along
+
+The beta release of Move 2024 comes with some powerful new features beyond the breaking changes described here. There are also more on the horizon. Join the Sui newsletter to learn about new, exciting features coming to Move this year, including syntactic macros, enums with pattern matching, and other user-defined syntax extensions.
+
+Please, provide any feedback or report any issues you encounter via GitHub, Discord, or Telegram.
```

Comment:
Not sure the best place to direct these.

Reply:
Yeah I just listed these out arbitrarily. I'll defer to others for the feedback links here (wherever you want to send 'em, let's send 'em there)